tools/autograd/derivatives.yaml

```diff
@@ -105,6 +113,38 @@
 #           marked as differentiable views and created with allow_rebase_history=false.
 #           Meaning that any inplace operation on it will raise an error. (unbind for example)
 #
+#  Notes about undefined output gradients:
+#     All backward functions must support all combinations of undefined output
+#     gradient Tensors, where `grad[i].defined() == false`. Code generation
+#     automatically adds this support to all derivative formulas that only
+#     calculate one input gradient, so in that case, you shouldn't have to think
```

Comment:
Isn't that when it only uses a single output gradient?

Reply:
Ah you're right. The codegen only applies to cases where there is only one input grad AND all output grads are undefined. If there's more than one output grad, the backward function needs to explicitly support a mixture of undefined/defined output grads. I'll fix the comment.